### PR TITLE
Log more information about exited children

### PR DESCRIPTION
**Pull request by nalind**
This logs more information, when available, about how the child processes that we reap terminated.

### CODE REVIEW

File: utils/utils.go [hunk 1]
```diff
@@ -144,14 +144,26 @@ func StartReaper() {
...
 			sig := <-sigs
-			for {
+			for sig == syscall.SIGCHLD {
 				// Reap processes
-				cpid, _ := syscall.Wait4(-1, nil, syscall.WNOHANG, nil)
+				var status syscall.WaitStatus
+				cpid, err := syscall.Wait4(-1, &status, syscall.WNOHANG, nil)
+				if err != nil {
+					if err != syscall.ECHILD {
+						logrus.Infof("wait4: %v", err)
```
Comment (runcom) [inline utils/utils.go]: should these stay at "debug" level?

Comment (nalind) [inline utils/utils.go]: Whoops.  Yes, almost certainly.

Comment (nalind) [inline utils/utils.go]: Fixed.

Comment (runcom) [inline utils/utils.go]: This very one seems to be `Infof` still :confused:

File: utils/utils.go [hunk 2]
```diff
@@ -144,14 +144,26 @@ func StartReaper() {
 		for {
 			// Wait for a child to terminate
 			sig := <-sigs
-			for {
+			for sig == syscall.SIGCHLD {
```
Comment (runcom) [inline utils/utils.go]: don't we drop all other sigs this way? should we have a switch inside the for loop

File: utils/utils.go (hunk 1 above)
Comment (nalind) [inline utils/utils.go]: Hmm, we expect ECHILD if we're calling wait4, but any other error result is unexpected.  Still, I guess we were just ignoring them all before, so :neutral_face:.  Changing it.

File: utils/utils.go (hunk 2 above)
Comment (nalind) [inline utils/utils.go]: Okay, change made.

**Comment by mrunalp** [comment]
LGTM

**Pull request merged**